src/services/services.ts

```diff
@@ -6107,7 +6229,7 @@ namespace ts {
                         cancellationToken.throwIfCancellationRequested();
 
                         const referenceLocation = getTouchingPropertyName(sourceFile, position);
-                        if (!isValidReferencePosition(referenceLocation, searchText)) {
+                        if (!implementations && !isValidReferencePosition(referenceLocation, searchText)) {

```

Comment:
do not think you need this. if is not valid reference you do not want to consider it for implementations anyways.